src/field/pokemon.ts

```diff
@@ -499,11 +500,19 @@ export abstract class Pokemon extends Phaser.GameObjects.Container {
   /**
    * Checks if a pokemon is fainted (ie: its `hp <= 0`).
    * It's usually better to call {@linkcode isAllowedInBattle()}
-   * @param checkStatus `true` to also check that the pokemon's status is {@linkcode StatusEffect.FAINT}
    * @returns `true` if the pokemon is fainted
    */
-  public isFainted(checkStatus: boolean = false): boolean {
-    return this.hp <= 0 && (!checkStatus || this.status?.effect === StatusEffect.FAINT);
+  public isFainted(): boolean {
+    return this.hp <= 0;
+  }
+
+  /**
+   * Faints the pokemon.
+   * @todo Handle all fainting-related side-effects here
+   */
+  public faint(): void {
+    this.hp = 0;
+    this.status = null;
```

Comment:
I think we should call this.resetStatus() instead, specifically because of this, which resets the Pokemon animation speed if the previous status was sleep
https://github.com/Despair-Games/poketernity/blob/d1b4f6d1a4bd811d7d3ad698c5a343acb5e4cbd3/src/field/pokemon.ts#L4166-L4167
Maybe also because of the confusion tag removal?
```suggestion
    this.resetStatus(true);
```

Reply:
Good point.